Module16/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,150 @@
+﻿#nullable disable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using Module16.Data;
+using Module16.Models;
+
+namespace Module16.Controllers
+{
+    public class ProductsController : Controller
+    {
+        private readonly NorthwindContext _context;
+        private readonly IConfiguration _configuration;
+
+        public ProductsController(NorthwindContext context, IConfiguration configuration)
```

Comment:
It is not a good practice to use Context inside Controller... Better would be to use Repository pattern